### PR TITLE
Add metrics for the BEAM-VM

### PR DESCRIPTION
**Pull request by RoadRunnr**
## Changes

Add metrics for the BEAM-VM

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [x] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

**Comment by RoadRunnr** [comment]
@tsloughter @ferd following the comment on https://github.com/open-telemetry/opentelemetry-erlang-contrib/pull/299, I've prepared a suggestion for the BEAM-VM metrics.

I think it makes sense to agree on a specification before continuing with the implementation.

File: docs/runtime/beam-metrics.md [hunk 1]
```diff
@@ -0,0 +1,947 @@
...
+| `bif` | Time spent in BIFs. Without extra states this time is part of the emulator state. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `busy_wait` | Time spent busy waiting. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `ets` | Time spent executing ETS BIFs. Without extra states this time is part of the emulator state. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `gc_full` | Time spent doing fullsweep garbage collection. Without extra states this time is part of the gc state. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `nif` | Time spent in NIFs. Without extra states this time is part of the emulator state. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `send` | Time spent sending messages (processes only). Without extra states this time is part of the emulator state. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `timers` | Time spent managing timers. Without extra states this time is part of the other state. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+<!-- endsemconv -->
```
Comment (ferd) [inline docs/runtime/beam-metrics.md:140]: We should probably repeat the microstate accounting caveat that seems to only exist in the declaration yaml and code for this section?

File: docs/runtime/beam-metrics.md [hunk 2]
```diff
@@ -0,0 +1,947 @@
...
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `beam.memory.allocators.carrier.size` | UpDownCounter | `By` | Total size of the memory carriers for the different allocators in the VM. See erts_alloc(3). | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+<!-- endsemconv -->
+
+<!-- semconv metric.beam.memory.allocators.carrier.size(full) -->
+| Attribute  | Type | Description  | Examples  | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) | Stability |
+|---|---|---|---|---|---|
+| `beam.memory.allocators.alloc` | string | The allocation type. | `sys_alloc` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `beam.memory.allocators.instance_no` | int | The logical instance number [0..n] | `1` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```
Comment (ferd) [inline docs/runtime/beam-metrics.md:679]: I feel like using `beam.memory.allocators.instance` here is not losing any descriptiveness but also lines up with what the OTP docs would provide?

**Comment by jsuereth** [comment]
@tsloughter would love a review on this.

**Comment by RoadRunnr** [comment]
@ferd I've addressed you comments, but I don't have rights to reopen the PR.

Could you check if the new version is acceptable to you and maybe reopen the PR?

**Comment by tsloughter** [comment]
@jsuereth sorry! I somehow missed this. I'll start reviewing (may take a bit) but can it be reopened, I'll get it done within the 7 day limit for sure :).

File: docs/runtime/beam-metrics.md (hunk 2 above)
Comment (tsloughter) [reply docs/runtime/beam-metrics.md:679]: agreed.

File: docs/runtime/beam-metrics.md (hunk 1 above)
Comment (tsloughter) [reply docs/runtime/beam-metrics.md:140]: ^ this

File: .chloggen/add_beam.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Use this changelog template to create an entry for release notes.
+#
+# If your change doesn't affect end users you should instead start
+# your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: new_component
+
+# The name of the area of concern in the attributes-registry, (e.g. http, cloud, db)
+component: beamvm
```
Comment (tsloughter) [inline .chloggen/add_beam.yaml:10]: Should just be `beam`?

File: .chloggen/add_beam.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Add metrics for the BEAM-VM (use by Erlang, Elixir and others)
```
Comment (tsloughter) [inline .chloggen/add_beam.yaml:13]: `note: Add metrics for the BEAM-VM (used by Erlang, Elixir and others)`

**Comment by tsloughter** [comment]
seems I can't make suggestions on a closed PR.

**Comment by RoadRunnr** [comment]
> seems I can't make suggestions on a closed PR.

**argh**, the PR also has not picked up the changes I pushed to my branch, probably because it's closed. So, most of the comments are already addressed.

@tsloughter I think it will be simpler if I open a new PR.

**Pull request closed without merging**